### PR TITLE
feat: flagsClient custom config

### PR DESCRIPTION
**Pull request by Netail**
## About the changes

Allow passing of custom clientKey, appName & url into the flagsClient instead of relying on env variables

### CODE REVIEW

**Comment by Netail** [comment]
Might be good to add a pre-commit hook for prettier. When I ran prettier, it formatted 34 files

File: lib/src/utils.ts [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const safeCompare = (a: string, b: string) => {
 export const randomSessionId = () =>
   `${Math.floor(Math.random() * 1_000_000_000)}`;
 
-const validateEnviromentVariables = () => {
```
Comment (Tymek) [inline lib/src/utils.ts:29]: Good catch 👍🏻

**Review by Tymek — COMMENTED** [review]
Very good addition. Would you like to also add `should send metrics` test, or some check that config is correct?

File: lib/src/utils.ts (hunk 1 above)
Comment (Netail) [reply lib/src/utils.ts:29]: VSCode `streetsidesoftware.code-spell-checker` :)

**Comment by Netail** [comment]
> Very good addition. Would you like to also add `should send metrics` test, or some check that config is correct?

Added addition tests regarding sending metrics

**Review by Tymek — APPROVED** [review]

**Comment by Tymek** [comment]
This can now be tested in [v1.6.0-beta.0](https://github.com/Unleash/unleash-client-nextjs/releases/tag/v1.6.0-beta.0)
or just `beta` tag

**Pull request merged**